android/src/test/java/com/onegini/mobile/sdk/reactnative/HandleRegistrationCallbackUseCaseTests.kt

```diff
@@ -0,0 +1,90 @@
+package com.onegini.mobile.sdk.reactnative
+
+import android.net.Uri
+import com.facebook.react.bridge.Promise
+import com.onegini.mobile.sdk.android.handlers.request.callback.OneginiBrowserRegistrationCallback
+import com.onegini.mobile.sdk.reactnative.clean.use_cases.HandleRegistrationCallbackUseCase
+import com.onegini.mobile.sdk.reactnative.exception.OneginiWrapperErrors.REGISTRATION_NOT_IN_PROGRESS
+import com.onegini.mobile.sdk.reactnative.facade.UriFacade
+import com.onegini.mobile.sdk.reactnative.handlers.registration.RegistrationEventEmitter
+import com.onegini.mobile.sdk.reactnative.handlers.registration.RegistrationRequestHandler
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.Mockito.spy
+import org.mockito.junit.MockitoJUnitRunner
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+
+@RunWith(MockitoJUnitRunner::class)
+
+class HandleRegistrationCallbackUseCaseTests {
+
+    @Mock
+    private lateinit var callbackMock: OneginiBrowserRegistrationCallback
+
+    @Mock
+    private lateinit var promiseMock: Promise
+
+    @Mock
+    private lateinit var uriMock: Uri
+
+    @Mock
+    private lateinit var parsedUri: Uri
+
+    @Mock
+    private lateinit var uriFacade: UriFacade
+
+    private lateinit var handleRegistrationCallbackUseCase: HandleRegistrationCallbackUseCase
```

Comment:
you can move this declaration a bit down, so first all mocks are present, below class under test with other non-mocks